metricbeat/module/docker/docker.go

```diff
@@ -84,6 +84,7 @@ func FetchStats(client *docker.Client) ([]Stat, error) {
 	}()
 
 	wg.Wait()
+	close(queue)
```

Comment:
And up on line 69, consider changing it to `containersList = make([]Stat, 0, len(containers))` since we know ahead of time how big the slice needs to be. This will reduce the number of allocations.